Design/ThirdpartyDependencies.md

```diff
@@ -0,0 +1,300 @@
+It would require some patching of `site.py` or using `sitecustomize.py`
+to ensure packaging doesn't interfere with system versions
+but this is something supported by Python
+and done by other systems like `Conda`.
+Python requirements could be easily supported by a requirements file
+or some newer system like
+[pipenv](https://packaging.python.org/key_projects/#pipenv) to properly
+define a frozen set of dependencies.
+Some packages such as `PyQt` do not provide the headers and link
+libraries required for the C++ build.
+The easiest solution would be a Conan package for `PyQt` much like macOS
+Homebrew does. 
+
+The availability of `mantid` to other Python libraries would be the cause
+of the most concern if we adopted the Conan approach as it is
+about creating a self-contained package.
+
+
+### Conda
+
+*"A conda package is a binary tarball containing
+system-level libraries,
+Python modules,
+executable programs,
+or other components"*.
+
+[Conda](https://docs.conda.io/en/latest/)
+is another cross-platform package manager but with support for a
+variety of languages including Python and C++.
+Alongside `pip` many popular Python packages support distribution through Conda.
+In fact `mantid` already has [Conda packages](https://anaconda.org/mantid/repo)
+for workbench and the framework for Linux and Mac.
+
+[Anaconda](https://repo.continuum.io/pkgs/) hosts many prebuilt packages for
+various combinations of compilers and operating systems.
+[Conda build](https://conda.io/projects/conda-build/en/latest/resources/commands/conda-build.html)
+is provided to enable building custom packages.
+
+The [meta.yaml](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#requirements-section)
+file provides a central place to define dependencies and would satisfy requirements 2-4.
+The requirements file allows builds tools such as CMake to be used.
+
+It is unclear how the build process would fit into current tools such as IDEs.
+The current CLI workflow is documented [here](https://github.com/mantidproject/conda-recipes/blob/master/Developer.md).
+
+Conda would possibly a better solution for providing just the `mantid` framework
+as a library for other users to import and not considering this as a solution
+for providing the full desktop application bundle. We also already have a
+working conda build of the framework so this is low overhead. This needs
+proper support though.
+
+## Summary + Questions
+
+After consideration I would suggest be that we adopt both options for supporting
+the two use cases presented. Both cases have different needs so there is no
+reason to expect one system to support both.
+
+Questions: 
+
+1. Do we need to still support a build against system libraries on Linux?
+
+   * Would SNS need an RPM framework build for their GUIs linked to system Python?
+
+2. Is it sensible to consider maintaining Conan packages as suggested?
+
+  * Where would we store binary artifacts? Or just store recipes?
```

Comment:
What would be the rough time to build locally per machine as an estimate?

If it's an hour per developer per say 6-12 months, configuring remote artefacts / storage / retention vs building locally would become a factor

Reply:
It is a very good question that I have no answer for at the moment. Most core framework/qt dependencies exist in artefact repos somewhere but this is an issue as our build time is long enough.